### PR TITLE
Add 'delete-instances' option to RDS Cluster 'delete' action

### LINKED ISSUE #371: Add RDS Cluster 'delete' action to first delete associated RDS DB instances
For the RDS Cluster resource added in https://github.com/capitalone/cloud-custodian/pull/344 it would be good to enhance the **delete** action to add an option (default `false`) to first delete any associated DB instances.

### PR DESCRIPTION
**Pull request by ewbankkit**
* Add unit tests and documentation for RDS Cluster 'delete'/'delete-instances' action option
* Also tidied up various Flake8 issues with `rdscluster.py`
* Fixes #371

### CODE REVIEW

File: c7n/resources/rdscluster.py [hunk 1]
```diff
@@ -54,27 +54,43 @@ class Meta(object):
...
     schema = type_schema(
-        'delete', **{'skip-snapshot': {'type': 'boolean'}})
+        'delete', **{'skip-snapshot': {'type': 'boolean'},
+                     'delete-instances': {'type': 'boolean'}})
 
-    def process(self, resources):
-        self.skip = self.data.get('skip-snapshot', False)
+    def process(self, clusters):
+        skip = self.data.get('skip-snapshot', False)
+        delete_instances = self.data.get('delete-instances', False)
```
Comment (kapilt) [inline c7n/resources/rdscluster.py]: should this default to True, it feels wierd to delete the cluster but leave all the instances bits (compared to an asg that left all its instances?) is this even nesc? do individual instances survive the cluster?

Comment (kapilt) [inline c7n/resources/rdscluster.py]: if we do need this feels like we should do it post deletion of the cluster itself, else how would you get a final snapshot if you have no instances.

Comment (ewbankkit) [inline c7n/resources/rdscluster.py]: The cluster itself cannot be deleted if any of its instances are not in deleted/deleting state (an `InvalidDBClusterStateFault` exception is raised).
Agreed that we should default to *True*.
I'll change (and rebase).

**Comment by ewbankkit** [comment]
Made changes and rebased.

**Comment by kapilt** [comment]
could update this pull request

**Comment by ewbankkit** [comment]
Will do...

**Comment by ewbankkit** [comment]
Rebased.

**Comment by kapilt** [comment]
lgtm

**Pull request merged**